app/src/main/java/io/apicurio/registry/store/RegistryStorage.java

```diff
@@ -0,0 +1,78 @@
+    
+    public void deleteArtifact(String artifactId) throws ArtifactNotFoundException, RegistryStorageException;
+    
+    public StoredArtifact getArtifact(String artifactId) throws ArtifactNotFoundException, RegistryStorageException;
+    
+    public ArtifactMetaData updateArtifact(String artifactId, String content) throws ArtifactNotFoundException, RegistryStorageException;
+    
+    public ArtifactMetaData getArtifactMetaData(String artifactId) throws ArtifactNotFoundException, RegistryStorageException;
+    
+    public void updateArtifactMetaData(String artifactId, EditableMetaData metaData) throws ArtifactNotFoundException, RegistryStorageException;
+    
+    public List<Rule> getArtifactRules(String artifactId) throws ArtifactNotFoundException, RegistryStorageException;
+    
+    public void createArtifactRule(String artifactId, Rule rule) throws ArtifactNotFoundException, ArtifactRuleAlreadyExistsException, RegistryStorageException;
+    
+    public void deleteArtifactRules(String artifactId) throws ArtifactNotFoundException, RegistryStorageException;
+    
+    public Rule getArtifactRule(String artifactId, String ruleName) throws ArtifactNotFoundException, RuleNotFoundException, RegistryStorageException;
+    
+    public void deleteArtifactRule(String artifactId, String ruleNam) throws ArtifactNotFoundException, RuleNotFoundException, RegistryStorageException;
+    
+    public List<Long> getArtifactVersions(String artifactId) throws ArtifactNotFoundException, RegistryStorageException;
+    
+    public StoredArtifact getArtifactVersion(String artifactId, long version) throws ArtifactNotFoundException, VersionNotFoundException, RegistryStorageException;
+    
+    public void deleteArtifactVersion(String artifactId, long version) throws ArtifactNotFoundException, VersionNotFoundException, RegistryStorageException;
+    
+    public VersionMetaData getArtifactVersionMetaData(String artifactId, long version) throws ArtifactNotFoundException, VersionNotFoundException, RegistryStorageException;
+    
+    public void updateArtifactVersionMetaData(String artifactId, long version, EditableMetaData metaData) throws ArtifactNotFoundException, VersionNotFoundException, RegistryStorageException;
+    
+    public void deleteArtifactVersionMetaData(String artifactId, long version) throws ArtifactNotFoundException, VersionNotFoundException, RegistryStorageException;
+    
+    public List<Rule> getGlobalRules() throws RegistryStorageException;
+    
+    public void createGlobalRule(Rule rule) throws RegistryStorageException;
+    
+    public void deleteGlobalRules() throws RegistryStorageException;
+    
+    public Rule getGlobalRule(String ruleName) throws RuleNotFoundException, RegistryStorageException;
+
+    public void updateGlobalRule(Rule rule) throws RuleNotFoundException, RegistryStorageException;
```

Comment:
There is no `updateArtifactRule`.

Reply:
Thanks - that's an oversight.  I'll fix before I merge.